besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1308,6 +1308,12 @@ private void issueOptionWarnings() {
         "--sync-mode",
         !SyncMode.FAST.equals(syncMode),
         singletonList("--fast-sync-min-peers"));
+
+    if (!securityModuleName.equals(DEFAULT_SECURITY_MODULE) && nodePrivateKeyFile != null) {
+      logger.warn(
+          "--node-private-key-file will have no effect unless --security-module={} is defined on the command line.",
```

Comment:
node-private-key-file also works if no security-module is specified so I think we need to mention that somehow. Maybe "--node-private-key-file will have no effect unless --security-module={} is defined on the command line or none is specified".

Reply:
this warning will kick if user is explicitly setting non-default security module value i.e.`--security-module=nondefault --node-private-key-file=/somepath`.